File: website/Moe.La.Web/els-app/src/app/core/models/workflow-type.ts
import { Guid } from "guid-typescript";

import { BaseModel } from "./base-model";
import { WorkflowActionListItem } from './workflow-action';
import { WorkflowStepListItem } from './workflow-step';
import { WorkflowStepActionListItem } from './workflow-step-action';

/**
 * Used to create workflow type objects.
 */
export class WorkflowType extends BaseModel<string> {
    constructor() {
        super();
        this.id = Guid.createEmpty().toString();
        this.createdBy = Guid.createEmpty().toString();
    }

    /**
     * The workflow type Arabic name.
     */
    typeArName: string = '';

    /**
     * Detemines wether the workflow type is active or not.
     */
    isActive: boolean = false;

    /**
     * Detemines the amount of time allowed for this workflow type to be locked by a given user.
     */
    lockPeriod: number = 0; // hours.
}

/**
 * Used to create workflow type list item objects.
 */
export class WorkflowTypeListItem {
    /**
     * ID
     */
    id: string = '';

    /**
     * The workflow type Arabic name.
     */
    typeArName: string = '';

    /**
     * Detemines wether the workflow type is active or not.
     */
    isActive: boolean = false;

    /**
     * Detemines the amount of time allowed for this workflow type to be locked by a given user.
     */
    lockPeriod: number = 0; // hours.
}

/**
 * A view model for workflow type component.
 */
export class WorkflowTypeView {
    /**
     * Workflow type.
     */
    workflowType: WorkflowType = null!;

    /**
     * Workflow actions.
     */
    workflowActions: WorkflowActionListItem[] = [];

    /**
     * Workflow steps.
     */
    workflowSteps: WorkflowStepListItem[] = [];

    /**
     * Workflow steps actions.
     */
    workflowStepsActions: WorkflowStepActionListItem[] = [];
}

File: services/src/Moe.La.Core/Entities/Workflow/WorkflowType.cs
using System;
using System.Collections.Generic;

namespace Moe.La.Core.Entities
{
    /// <summary>
    /// The workflow type class.
    /// </summary>
    public class WorkflowType : BaseEntity<Guid>
    {
        /// <summary>
        /// The workflow type Arabic name.
        /// </summary>
        public string TypeArName { get; set; }

        /// <summary>
        /// Detemines wether the workflow type is active or not.
        /// </summary>
        public bool IsActive { get; set; }

        /// <summary>
        /// Detemines the amount of time allowed for this workflow type to be locked by a given user.
        /// </summary>
        public int LockPeriod { get; set; }

        /// <summary>
        /// The workflow steps navigation property.
        /// </summary>
        public virtual ICollection<WorkflowStep> WorkflowSteps { get; set; } = new List<WorkflowStep>();

        /// <summary>
        /// The workflow actions navigation property.
        /// </summary>
        public virtual ICollection<WorkflowAction> WorkflowActions { get; set; } = new List<WorkflowAction>();

        /// <summary>
        /// The workflow instances navigation property.
        /// </summary>
        public virtual ICollection<WorkflowInstance> WorkflowInstances { get; set; } = new List<WorkflowInstance>();
    }
}
